addons/project/views/project_sharing_views.xml

```diff
@@ -239,7 +239,7 @@
                                     <field name="stage_id" optional="show"/>
                                     <button name="action_open_task" type="object" title="View Task" string="View Task" class="btn btn-link float-end"
                                             context="{'form_view_ref': 'project.project_sharing_project_task_view_form'}"
-                                            attrs="{'invisible': &quot;[('display_project_id', '!=', False), ('display_project_id', '!=', active_id)]&quot;}"/>
```

Comment:
Hello, I've updated the required changes. Please take a look, and let me know if there's anything else that requires updating. Thank you!

Reply:
I suggested changing `display_project_id` for `project_id`. Was it a bad idea?
If so, could you explain why?

Reply:
Hello @auon-odoo it is not the bad idea to apply the check on on the project_id but the issue i am facing is when i create new subtask and click on add line by default there will be current project should be added but in display_project_id field it will not added current project so i try all the checks on the project_id but haven't found an ideal solution yet. because of this issue i apply checks on the display_project_id open to suggestions!

Reply:
I pushed my suggestion. I tested it and it seems to work just fine.